sunpy/net/dataretriever/sources/goes.py

```diff
@@ -58,18 +58,22 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             Data type to return for the particular GOES satellite. Supported
             types depend on the satellite number specified. (default = xrs_2s)
         """
-
         # find out which satellite and datatype to query from the query times
         sat_num = self._get_goes_sat_num(timerange.start, timerange.end)
         base_url = 'http://umbra.nascom.nasa.gov/goes/fits/'
-
-        if timerange.start < parse_time('1999/01/15'):
-            url = base_url + "{date:%Y}/go{sat:02d}{date:%y%m%d}.fits".format(
-                date=timerange.start, sat=sat_num[0])
-        else:
-            url = base_url + "{date:%Y}/go{sat:02d}{date:%Y%m%d}.fits".format(
-                date=timerange.start, sat=sat_num[0])
-        return [url]
+        total_days = (timerange.end - timerange.start).days + 1 
+        all_dates = timerange.split(total_days)
+        result = list()
+        for day in all_dates:
+            regex = "{date:%Y}/go{sat:02d}"
+            if (day.end < parse_time('1999/01/15')):
+                regex += "{date:%y%m%d}.fits"
+            else:
+                regex += "{date:%Y%m%d}.fits"
+            url = base_url + regex.format(
+                    date=day.end, sat=self._get_goes_sat_num(day.end, day.end)[0])
```

Comment:
All right, that makes sense. 
Could you write that into the source code as a comment? so future me won't ask again :)